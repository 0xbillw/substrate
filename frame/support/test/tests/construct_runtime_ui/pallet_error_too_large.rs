use frame_support::construct_runtime;
use sp_runtime::{generic, traits::BlakeTwo256};
use sp_core::sr25519;

#[frame_support::pallet]
mod pallet {
	#[pallet::config]
	pub trait Config: frame_system::Config {}

	#[pallet::pallet]
	pub struct Pallet<T>(core::marker::PhantomData<T>);

	#[pallet::error]
	pub enum Error<T> {
		MyError(crate::Nested1),
	}
}

#[derive(scale_info::TypeInfo, frame_support::PalletError, codec::Encode, codec::Decode)]
pub enum Nested1 {
	Nested2(Nested2)
}

#[derive(scale_info::TypeInfo, frame_support::PalletError, codec::Encode, codec::Decode)]
pub enum Nested2 {
	Nested3(Nested3)
}

#[derive(scale_info::TypeInfo, frame_support::PalletError, codec::Encode, codec::Decode)]
pub enum Nested3 {
	Nested4(Nested4)
}

#[derive(scale_info::TypeInfo, frame_support::PalletError, codec::Encode, codec::Decode)]
pub enum Nested4 {
	Num(u8)
}

pub type Signature = sr25519::Signature;
pub type BlockNumber = u64;
pub type Header = generic::Header<BlockNumber, BlakeTwo256>;
pub type Block = generic::Block<Header, UncheckedExtrinsic>;
pub type UncheckedExtrinsic = generic::UncheckedExtrinsic<u32, Call, Signature, ()>;

impl pallet::Config for Runtime {}

construct_runtime! {
	pub enum Runtime where
		Block = Block,
		NodeBlock = Block,
		UncheckedExtrinsic = UncheckedExtrinsic
	{
		System: system::{Pallet, Call, Storage, Config, Event<T>},
		Pallet: pallet::{Pallet},
	}
}

fn main() {}
