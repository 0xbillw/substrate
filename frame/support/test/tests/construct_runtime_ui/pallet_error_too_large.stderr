error[E0433]: failed to resolve: use of undeclared crate or module `system`
  --> tests/construct_runtime_ui/pallet_error_too_large.rs:53:11
   |
53 |         System: system::{Pallet, Call, Storage, Config, Event<T>},
   |                 ^^^^^^ use of undeclared crate or module `system`

error[E0433]: failed to resolve: use of undeclared crate or module `system`
  --> tests/construct_runtime_ui/pallet_error_too_large.rs:47:1
   |
47 | / construct_runtime! {
48 | |     pub enum Runtime where
49 | |         Block = Block,
50 | |         NodeBlock = Block,
...  |
55 | |     }
56 | | }
   | |_^ not found in `system`
   |
   = note: this error originates in the macro `construct_runtime` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing this enum
   |
1  | use frame_system::RawOrigin;
   |

error[E0433]: failed to resolve: use of undeclared crate or module `system`
  --> tests/construct_runtime_ui/pallet_error_too_large.rs:47:1
   |
47 | / construct_runtime! {
48 | |     pub enum Runtime where
49 | |         Block = Block,
50 | |         NodeBlock = Block,
...  |
55 | |     }
56 | | }
   | |_^ not found in `system`
   |
   = note: this error originates in the macro `construct_runtime` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
   |
1  | use crate::pallet::Pallet;
   |
1  | use frame_support_test::Pallet;
   |
1  | use frame_system::Pallet;
   |
1  | use test_pallet::Pallet;
   |

error[E0277]: the trait bound `Runtime: frame_system::Config` is not satisfied
  --> tests/construct_runtime_ui/pallet_error_too_large.rs:45:6
   |
45 | impl pallet::Config for Runtime {}
   |      ^^^^^^^^^^^^^^ the trait `frame_system::Config` is not implemented for `Runtime`
   |
note: required by a bound in `pallet::Config`
  --> tests/construct_runtime_ui/pallet_error_too_large.rs:8:20
   |
8  |     pub trait Config: frame_system::Config {}
   |                       ^^^^^^^^^^^^^^^^^^^^ required by this bound in `pallet::Config`

error[E0080]: evaluation of constant value failed
  --> tests/construct_runtime_ui/pallet_error_too_large.rs:47:1
   |
47 | / construct_runtime! {
48 | |     pub enum Runtime where
49 | |         Block = Block,
50 | |         NodeBlock = Block,
...  |
55 | |     }
56 | | }
   | |_^ attempt to compute `0_usize - 1_usize`, which would overflow
   |
   = note: this error originates in the macro `self::sp_api_hidden_includes_construct_runtime::hidden_include::const_assert` (in Nightly builds, run with -Z macro-backtrace for more info)
